src/main/java/com/squareup/javapoet/CodeWriter.java

```diff
@@ -401,8 +401,16 @@ String lookupName(ClassName className) {
 
     // If the class is in the same package, we're done.
     if (Objects.equals(packageName, className.packageName())) {
-      referencedNames.add(topLevelSimpleName);
-      return join(".", className.simpleNames());
+      String conflictClassName = packageName+"."+className.simpleName;//From same package (Nest)
```

Comment:
Thanks for the fix! but what do you need this line for? It is already checked that `packageName ` equal to `className.packageName()`, so `packageName+"."+className.simpleName` is equivalent to `className.packageName() +"."+className.simpleName`, right? Then you don't need the comparison to canonicalName below, because it is always true

Reply:
Well, actually the `conflictClassName `is not always equal to the `className.canonicalName`, because of the existence of super class. The ignorance of trim only occur when the class is `alwaysQualify `and `canonicalName `equals, that is ,the class share a same name with a subclass but canonical name doesn't act like that. For more detail about the issue described situation, you can go to the `avoidClashesWithNestedClasses_SamePackageTest` in `JavaFileTest`

Reply:
> Thanks for the fix! but what do you need this line for? It is already checked that `packageName ` equal to `className.packageName()`, so `packageName+"."+className.simpleName` is equivalent to `className.packageName() +"."+className.simpleName`, right? Then you don't need the comparison to canonicalName below, because it is always true

Well, actually the `conflictClassName `is not always equal to the `className.canonicalName`, because of the existence of super class. The ignorance of trim only occur when the class is `alwaysQualify `and `canonicalName `equals, that is ,the class share a same name with a subclass but canonical name doesn't act like that. For more detail about the issue described situation, you can go to the `avoidClashesWithNestedClasses_SamePackageTest` in `JavaFileTest`